### ISSUE TITLE
Research: Evaluate RT-DETR vs YOLOX for wildlife detection

### ISSUE THREAD

**Issue by filthyrake** [filthyrake/telescope_cam_detection#163]
## Overview
Research and benchmark RT-DETR (transformer-based detector) as potential replacement for YOLOX.

## Motivation
2024-2025 research shows RT-DETR now matches/exceeds YOLO family:
- **RT-DETR-R50**: 53.1% AP @ 108 FPS (T4 GPU)
- **RT-DETRv2**: 55.3% AP @ 100 FPS (+2.2% AP over v1)
- **YOLOX-x**: 51.2% AP @ 68 FPS

Key advantages for wildlife detection:
- Better at **small objects** (transformers capture global context)
- More robust to **occlusion** (trees, bushes, partial visibility)
- Better **IR/night vision** performance (less sensitive to image quality)

## Research Questions
1. Does RT-DETR perform better on small/distant wildlife at 1920x1920?
2. How does TensorRT-optimized RT-DETR compare to YOLOX in speed?
3. Is accuracy gain worth potential migration effort?
4. How well does it handle IR/night vision footage?

### Phase 1: Offline Benchmarking
1. Download RT-DETR pretrained weights
2. Run on 500 annotated wildlife images (our existing dataset)
3. Compare metrics:
   - mAP (overall accuracy)
   - Small object detection (<32px)
   - Inference time (PyTorch and TensorRT)
   - False positive rate

### Phase 2: Live A/B Testing
1. Deploy RT-DETR alongside YOLOX
2. Run both models on same video stream
3. Compare detections side-by-side
4. Measure:
   - Detection counts
   - Confidence distributions
   - GPU utilization
   - User feedback (which misses fewer animals?)

### Step 1: Install and Test
```bash
pip install ultralytics

# Test RT-DETR
from ultralytics import RTDETR
model = RTDETR("rtdetr-x.pt")
results = model("path/to/wildlife.jpg")
```

### Step 2: Convert to TensorRT
```python
# Export to TensorRT for fair comparison
model.export(format="engine", imgsz=1920, half=True)
```

### Step 3: Benchmark Script
```python
# benchmark_models.py
import time
from ultralytics import RTDETR, YOLO

# Load models
yolox = YOLO("yolox-s.pt")
rtdetr = RTDETR("rtdetr-x.pt")

# Benchmark on test set
results = compare_models(
    models=[yolox, rtdetr],
    test_images="tests/wildlife_dataset/",
    metrics=["mAP", "fps", "small_object_recall"]
)
```

### Step 4: Integration (If Successful)
- Add RT-DETR option to config
- Update `YOLOXDetector` → generic `ObjectDetector`
- Support both models for migration period

## Success Criteria
RT-DETR is worth migrating if:
- [ ] ≥5% better mAP on wildlife test set
- [ ] ≥2x better small object detection (<32px)
- [ ] Comparable speed when TensorRT-optimized (<20% slower)
- [ ] Better night vision performance (subjective evaluation)

## Risks
- **Migration effort**: May require refactoring detection pipeline
- **Model size**: RT-DETR may be larger (more VRAM)
- **Unknown behavior**: Less proven for wildlife vs. COCO objects
- **Compatibility**: May not work with existing two-stage pipeline

## Timeline
- Week 1: Offline benchmarking
- Week 2: TensorRT optimization comparison
- Week 3: Live A/B testing
- Week 4: Decision and migration plan (if positive)

## Deliverables
- [ ] Benchmark report (mAP, FPS, small object performance)
- [ ] Annotated comparison videos (side-by-side detections)
- [ ] Migration effort estimate
- [ ] Recommendation: migrate, stay with YOLOX, or hybrid approach

## Priority
**MEDIUM** - Research task, high potential but uncertain outcome

## Related Issues
- Depends on TensorRT infrastructure (#157)
- May replace YOLOX if successful

## References
- RT-DETR paper: https://arxiv.org/abs/2304.08069 "DETRs Beat YOLOs on Real-time Object Detection"
- Ultralytics docs: https://docs.ultralytics.com/models/rtdetr/
- Comparison: https://docs.ultralytics.com/compare/rtdetr-vs-yolox/
